From d3eb4a11acec05314e45d4b78e6c0fb7c38a6fcd Mon Sep 17 00:00:00 2001
From: Evangelos Foutras <evangelos@foutrelis.com>
Date: Wed, 17 Jun 2020 17:42:45 +0300
Subject: [PATCH] LinGui: tweak About dialog so Close button works

Clicking on "Close" emits a "response" signal which was ignored. Make
the button work by calling gtk_dialog_run(), as it's done for similar
dialogs in callbacks.c.

Fixes https://github.com/HandBrake/HandBrake/issues/2405
---
 gtk/src/callbacks.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/gtk/src/callbacks.c b/gtk/src/callbacks.c
index 322cf0af3f..5cab0cb493 100644
--- a/gtk/src/callbacks.c
+++ b/gtk/src/callbacks.c
@@ -4542,7 +4542,8 @@ about_action_cb(GSimpleAction *action, GVariant *param, signal_user_data_t *ud)
                                 HB_PROJECT_URL_WEBSITE);
     gtk_about_dialog_set_website_label(GTK_ABOUT_DIALOG(widget),
                                         HB_PROJECT_URL_WEBSITE);
-    gtk_widget_show (widget);
+    gtk_dialog_run(GTK_DIALOG(widget));
+    gtk_widget_hide(widget);
 }
 
 #define HB_DOCS "https://handbrake.fr/docs/"
